provisioning/transport/amqp/src/sasl_tpm.ts

```diff
@@ -8,45 +8,56 @@ import * as Promise from 'bluebird';
+export type GetSasTokenCallback = (err: Error, sasToken?: string) => void;
+export type GetSasToken = (challenge: Buffer, callback: GetSasTokenCallback) => void;
 export type SaslResponseFrame = {
   response: Buffer
 };
 
 export class SaslTpm {
   public name: string = 'TPM';
 
-  private _init: Buffer;
-  private _firstResponse: Buffer;
-  private _getResponseFromChallenge: GetResponseFromChallenge;
+  private _getSasToken: GetSasToken;
   private _challengeKey: Builder;
-  private _hostname: string;
+  private _idScope: string;
+  private _registrationId: string;
+  private _endorsementKey: Buffer;
+  private _storageRootKey: Buffer;
 
   /*Codes_SRS_NODE_PROVISIONING_AMQP_SASL_TPM_18_001: [ The `SaslTpm` constructor shall accept the following parameters:
-    `hostName` - The hostName value to be returned when getInitFrame is called
-    `init` - The initial frame contents to be returned when getInitFrame is called
-    `firstResponse` - The response to return on the first call to getResponseFrame
-    `getResponseFromChallenge` - The callback to call when the challenge has been completed and the caller needs to formulate the response. ] */
-  constructor(hostName: string, init: Buffer, firstResponse: Buffer, getResponseFromChallenge: GetResponseFromChallenge) {
-    this._hostname = hostName;
-    this._init = init;
-    this._firstResponse = firstResponse;
-    this._getResponseFromChallenge = getResponseFromChallenge;
+    `idScope` - the idScope for the provisioning service instance
+    `registrationId` - the registrationId for the device being registered
+    `endorsementKey` - the endorsement key which was acquired from the TPM
+    `storageRootKey` - the storage root key which was acquired from the TPM
+    `getSasToken` - The callback to call when the challenge has been completed and the caller needs to formulate the response. ] */
+  constructor(idScope: string, registrationId: string, endorsementKey: Buffer, storageRootKey: Buffer, getSasToken: GetSasToken) {
+    this._idScope = idScope;
+    this._registrationId = registrationId;
+    this._endorsementKey = endorsementKey;
+    this._storageRootKey = storageRootKey;
+    this._getSasToken = getSasToken;
     this._challengeKey = new Builder();
-    this._hostname = hostName;
   }
 
   /*Codes_SRS_NODE_PROVISIONING_AMQP_SASL_TPM_18_002: [ `getInitFrame` shall return a promise that resolves to an object with the following members:
     `mechanism` - Must be 'TPM'
     `initialResponse` - The inital frame contents
     `hostname` - the hostName ] */
   getInitFrame(): Promise<any> {
+    let init: Buffer = new Builder()
+        .appendUInt8(0)
+        .appendString(this._idScope)
+        .appendUInt8(0)
+        .appendString(this._registrationId)
+        .appendUInt8(0)
+        .appendBuffer(this._endorsementKey)
+        .get();
+
     return new Promise((resolve: any) => {
```

Comment:
Since there is no way this fails, you can just write it like so:
```js
return Promise.resolve(<the response object>);
```
and you don't have to create a new promise that way